builder/evaluator.go

```diff
@@ -136,7 +138,13 @@ func (b *Builder) Run(context io.Reader) (string, error) {
 		}
 	}()
 
-	filename := path.Join(b.contextPath, "Dockerfile")
+	filename := path.Join(b.contextPath, b.dockerfileName)
+
+	absContextPath, _ := filepath.Abs(b.contextPath)
```

Comment:
Just curious, why you not checking errors here? :) You checked them on client side.